probe-rs/src/config/registry.rs

```diff
@@ -396,13 +389,22 @@ pub fn add_target_from_yaml<R>(yaml_reader: R) -> Result<(), RegistryError>
 where
     R: Read,
 {
-    REGISTRY.lock().unwrap().add_target_from_yaml(yaml_reader)
+    REGISTRY.lock().add_target_from_yaml(yaml_reader)
+}
+
+/// Get a list of all families which are contained in the internal
+/// registry.
+///
+/// As opposed to `families()` this function does not clone the families, but using it is
+/// slightly more cumbersome.
+pub fn families_ref() -> impl Deref<Target = [ChipFamily]> {
+    MutexGuard::map(REGISTRY.lock(), |registry| registry.families.as_mut_slice())
```

Comment:
Why is it a mut slice?

Reply:
Because it [needs to be mut](https://docs.rs/parking_lot/latest/parking_lot/type.MutexGuard.html#method.map)

Reply:
Funnily enough, using a mutex here is wrong, so the next iteration won't need `mut` 🥲